pkg/generate/app/cmd/resolve.go

```diff
@@ -108,7 +106,10 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 	for _, repo := range repositories {
 		repo.SetContextDir(g.ContextDir)
 	}
-
+	// Environment File where the parameters are stored
+	if len(g.EnvironmentFile) != 0 && !app.IsPossibleEnvironmentFile(g.EnvironmentFile) {
+		return nil, fmt.Errorf("when --env-file is specified you must provide a valid env file location")
+	}
```

Comment:
The strategy and other parameters had validations here in this file, so I have put it here. If  you think the validation should go into GetEnvironmentVars, I can put this there as well.